### PR TITLE
docs: move resource schema to website page

### PR DESCRIPTION
**Pull request by mattrosno**
Moves schema docs from [GitHub](https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md) to website so resource maintainers can reference the schema, and so we can cross-link to definitions from throughout the website.

No schema content changes, only added the page description, anchor links, and code snippet titles.

#### Changelog

**New**

- `/pages/contributing/schema.mdx`

**Changed**

- LHN `Contribute` to `Contributing` to be like "Designing" and "Developing" (all activities)
- (I don't think we need to worry about redirects since we're still beta.)

**Removed**

- `/docs/resource-schema.md` page

#### Testing / reviewing

image

### CODE REVIEW

**Comment by mattrosno** [comment]
@aubrey-oneal can you review the "Contribute" to "Contributing" LHN change? The intent with this change is that contributing is an activity, just like designing and developing is an activity.

**Review by francinelucca — CHANGES_REQUESTED** [review]
Looks good! 😍  We should update these links as welll? 
image

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [inline services/web-app/pages/contributing/schema.mdx:19]: I'd suggest putting the yml example underneath the install code snippet for the stable schema. Otherwise it feels like the example should have an anchor link.

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [inline services/web-app/pages/contributing/schema.mdx:28]: I think we could expand the text here a little to have a softer introduction. Perhaps adding something like:

Maintainers can control content and resource visibility through a metadata file that follows a clear schema. This helps us document each resource in a standardized way and enables everything to be indexed by Carbon’s new underlying platform. Below is the latest stable version of the schema.

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [inline services/web-app/pages/contributing/schema.mdx:213]: Suggest giving examples of assets for a softer intro: Assets include re-usable resources such as components, patterns, functions, and templates. Each asset has code, designs, and usage guidance.

(feel free to cut that last sentences as I know it's not always true).

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [inline services/web-app/pages/contributing/schema.mdx:281]: can we move Asset type above Asset keys? I get that when defining the new object you will want to fill out the keys but as a casual reader it feels easier to start with 'type' to get a grasp on what assets can be.

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [inline services/web-app/pages/contributing/schema.mdx:217]: It's unusual to have this second layer of anchor links-- for this version I think it's a good start. Let's consider addressing deep navigation later.

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [inline services/web-app/pages/contributing/schema.mdx:434]: For the intro text to design kits, Morgana was suggesting we mention specific tools. i.e. Design Kits are libraries from Figma and Sketch—and other tools such as Adobe XD and Invision. They are the design counterpart to coded assets and contain components and elements as well as wireframes and guidance to aid designers. Design kits are indexed in the [carbon-platform](https://github.com/carbon-design-system/carbon-platform) repository 
[carbon.yml](https://github.com/carbon-design-system/carbon-platform/blob/main/packages/resources/carbon.yml)
 data file.

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [inline services/web-app/pages/contributing/schema.mdx:120]: Suggest starting off defining libraries with the noun (what is is) rather than the verb (what can I do with it).

Libraries are 1:1 with code packages. All coded components, elements, patterns, or functions belong to a library and have a maintainer. Design kits with compatible code also live in libraries.

File: services/web-app/pages/contributing/schema.mdx
Comment (aubrey-oneal) [reply services/web-app/pages/contributing/schema.mdx:19]: I'm also wondering if this header should be 'Get started' -- something actionable.

**Review by aubrey-oneal — COMMENTED** [review]
Functionally everything is good to go (i.e. all links work). All of my comments should be taken as content suggestions that you can take or leave.

File: services/web-app/pages/contributing/schema.mdx
Comment (mattrosno) [reply services/web-app/pages/contributing/schema.mdx:281]: I think that would break our documentation pattern where we show all available keys, then below we describe each key in more detail when necessary.

**Comment by mattrosno** [comment]
@aubrey-oneal I incorporated your suggestion to move the first example lower in the page. I think we should review the content in a separate PR to not hold this issue up (e.g. refining how we define each resource, etc.)

**Review by francinelucca — APPROVED** [review]
LGTM!

**Comment by mattrosno** [comment]
@andreancardona can you review?

**Comment by mattrosno** [comment]
Shared over Slack, @aubrey-oneal approved the "Contribute" to "Contributing" change in side nav.

**Review by jharvey10 — APPROVED** [review]

**Pull request merged**
